src/RuleWithOptionsInterface.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+/**
+ * An extended version of {@see RuleInterface} which allows exporting a rule options in addition to a rule name. It's
+ * useful for passing to frontend for further identification and implementing client-side validation. If you don't need
+ * that (for example for REST API), use {@see RuleInterface} instead.
+ */
+interface RuleWithOptionsInterface extends RuleInterface
+{
+    /**
+     * Gets a rule options as associative array. It's used for identification on the frontend with further implementing
+     * of client-side validation. Usually it's just a mapping between rule property names and values.
+     *
+     * For a example, for this rule:
+     *
+     * ```php
+     * new SomeRule(property1: 'value1', property2: 'value2');
+     * ```
+     *
+     * the options will be:
+     *
+     * ```php
+     * [
+     *     'property1' => $this->>property1,
+     *     'property2' => $this->>property2,
+     *     // ...
+     * ];
+     * ```
+     *
+     * Sometimes it's helpful to include dynamic options as well:
```

Comment:
What is dynamic options?

Reply:
For example:

https://github.com/yiisoft/validator/blob/f4e4de303fb05877af5aa6716d04dce579939cca/src/Rule/Compare.php#L169-L172

https://github.com/yiisoft/validator/blob/f4e4de303fb05877af5aa6716d04dce579939cca/src/Rule/Compare.php#L138-L148

The idea was to show that not only properties can be included as options.

Reply:
May be completely remove this block? It's base PHP availabilities.

Reply:
OK. Removed it completely.